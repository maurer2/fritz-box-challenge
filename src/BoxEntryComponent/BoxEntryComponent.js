import React, { PureComponent } from 'react';
import styled from 'styled-components/macro';
import PropTypes from 'prop-types';

const BoxEntryWrapper = styled.div`
  display: block;
  margin-top: 1rem;
  margin-bottom: 1rem;
`;

const BoxEntryTitle = styled.h3`
  margin-top: 0;
  margin-bottom: 0;
`;

const BoxEntryValue = styled.span`
  white-space: nowrap;
`;

class BoxEntryComponent extends PureComponent {
  render() {
    return (
      <BoxEntryWrapper>
        <BoxEntryTitle>
          { this.props.entry.charAt(0).toUpperCase() + this.props.entry.slice(1) }
        </BoxEntryTitle>
        <BoxEntryValue>
        { this.props.value }
        </BoxEntryValue>
      </BoxEntryWrapper>
    );
  }
}

BoxEntryComponent.propTypes = {
  entry: PropTypes.string,
  value: PropTypes.string,
};

export default BoxEntryComponent;
