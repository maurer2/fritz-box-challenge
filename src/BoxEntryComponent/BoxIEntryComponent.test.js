import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import BoxEntryComponent from './BoxEntryComponent';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

describe('BoxEntryComponent', () => {
  const wrapper = shallow(<BoxEntryComponent entry="model" value="test" />);
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
