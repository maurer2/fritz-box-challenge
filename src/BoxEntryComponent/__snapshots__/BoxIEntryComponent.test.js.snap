// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoxEntryComponent should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BoxEntryComponent
    entry="model"
    value="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Model
        </ForwardRef>,
        <ForwardRef>
          test
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Model",
        },
        "ref": null,
        "rendered": "Model",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "BoxEntryTitle-sc-16sq9z3-1",
            "isStatic": true,
            "rules": Array [
              "margin-top:0;margin-bottom:0;font-weight:bold;",
            ],
          },
          "displayName": "BoxEntryTitle",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "BoxEntryTitle-sc-16sq9z3-1",
          "target": "dt",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "test",
        },
        "ref": null,
        "rendered": "test",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "BoxEntryValue-sc-16sq9z3-2",
            "isStatic": true,
            "rules": Array [
              "margin:0;",
            ],
          },
          "displayName": "BoxEntryValue",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "BoxEntryValue-sc-16sq9z3-2",
          "target": "dd",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "BoxEntryWrapper-sc-16sq9z3-0",
        "isStatic": true,
        "rules": Array [
          "display:block;margin:1rem;",
        ],
      },
      "displayName": "BoxEntryWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "BoxEntryWrapper-sc-16sq9z3-0",
      "target": "dl",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Model
          </ForwardRef>,
          <ForwardRef>
            test
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Model",
          },
          "ref": null,
          "rendered": "Model",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "BoxEntryTitle-sc-16sq9z3-1",
              "isStatic": true,
              "rules": Array [
                "margin-top:0;margin-bottom:0;font-weight:bold;",
              ],
            },
            "displayName": "BoxEntryTitle",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "BoxEntryTitle-sc-16sq9z3-1",
            "target": "dt",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "test",
          },
          "ref": null,
          "rendered": "test",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "BoxEntryValue-sc-16sq9z3-2",
              "isStatic": true,
              "rules": Array [
                "margin:0;",
              ],
            },
            "displayName": "BoxEntryValue",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "BoxEntryValue-sc-16sq9z3-2",
            "target": "dd",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "BoxEntryWrapper-sc-16sq9z3-0",
          "isStatic": true,
          "rules": Array [
            "display:block;margin:1rem;",
          ],
        },
        "displayName": "BoxEntryWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "BoxEntryWrapper-sc-16sq9z3-0",
        "target": "dl",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
